src/components/Contacts/ContactsContext/ContactsContext.tsx

```diff
@@ -151,20 +150,16 @@ export const ContactsProvider: React.FC<ContactsContextProps> = ({
     [activeFilters],
   );
 
-  // Load the user's view preference
-  const { loading: userOptionsLoading } = useGetUserOptionsQuery({
-    // This fetch policy ensures that our onCompleted callback will only be called once instead of
-    // being called every time the cached user options are updated by the updateUserOptions mutation.
-    // nextFetchPolicy: 'standby',
-    onCompleted: ({ userOptions }) => {
-      const option = userOptions.find(
-        (option) => option.key === 'contacts_view',
-      );
-      if (option && option.value) {
-        setViewMode(option.value as TableViewModeEnum);
-      }
-    },
-  });
+  const [contactsView, updateOptions, { loading: userOptionsLoading }] =
+    useUserPreference({
+      key: 'contacts_view',
+      defaultValue: TableViewModeEnum.List,
+    });
+  useEffect(() => {
+    if (contactsView) {
```

Comment:
I think we shouldn't call `setViewMode` if `userOptionsLoading === true` because otherwise we might be trying to save the default value to the server.

Reply:
Gotcha, I will change it to ensure userOptions have finished loading is false